src/components/PrimaryPanes/SourcesTree.js

```diff
@@ -48,24 +48,25 @@ import { features } from "../../utils/prefs";
 import { setProjectDirectoryRoot } from "../../actions/ui";
 
 type Props = {
+  debuggeeUrl: string,
+  expanded?: any,
+  projectRoot: string,
   selectLocation: Object => void,
+  selectedSource?: SourceRecord,
   setExpandedState: any => void,
-  sources: SourcesMap,
   shownSource?: string,
-  selectedSource?: SourceRecord,
-  debuggeeUrl: string,
-  projectRoot: string,
-  expanded?: any
+  sources: SourcesMap
 };
 
 type State = {
   focusedItem?: any,
+  highlightItems?: any,
+  isSourceTreeLoaded: boolean,
```

Comment:
Seems better to use `treeProps.key == empty` for checking if there is anything in the tree instead of introducing a new property in state.